runtime/gc_base/StringTable.cpp

```diff
@@ -524,6 +524,42 @@ j9gc_stringHashFn(void *key, void *userData)
 	return stringHashFn(key, userData);
 }
 
+/**
+ * Assuming incoming UTF8 byte stream contains only ISO-8859-1/Latin-1 characters.
+ * Decode data & store to byteArray.
+ * 
+ * @param vmThread[in] the VM thread
+ * @param data[in] the UTF8 byte stream
+ * @param length[in] the length of incoming UTF8 byte stream
+ * @param byteArray[out] the target byte array to store the data
+ * @param translateSlashes[in] the flag to translate slashes
+ * 
+ * @return last slash position to be restored to `/` for anonymous classes
+ */
+MMINLINE static UDATA
+storeLatin1ByteArrayhelper(J9VMThread *vmThread, U_8 *data, UDATA length, j9object_t byteArray, bool translateSlashes) {
+	UDATA lastSlash = 0;
+	U_8 *dataTmp = data;
+	UDATA lengthTmp = length;
+	UDATA index = 0;
+	while (0 != lengthTmp) {
+		U_16 unicode = 0;
+		UDATA consumed = VM_VMHelpers::decodeUTF8CharN(dataTmp, &unicode, lengthTmp);
+		if (0 == consumed) {
+			Assert_MM_unreachable();
+		}
+		dataTmp += consumed;
+		lengthTmp -= consumed;
```

Comment:
If it possible (not too much performance impact) I would add check `(consumed <= lengthTmp)` before subtraction and assert otherwise. `lengthTmp` is declared unsigned so it might be wrapped around and negative result treated as very large positive. This might be a reaction (for example) to garbage passed and make execution go out of control

Reply:
The assumption here is that the input data is valid, so even the assert is somewhat unnecessary. Also note that the tmp variables are unnecessary (the args can be modified in place).

Reply:
Removing the tmp variables and use the args instead.

Reply:
Updated the assert like https://github.com/eclipse/openj9/blob/8317935a0b1d067d4a67b8c0378a20dfcbf4d87e/runtime/gc_base/StringTable.cpp#L626